src/frontend/src/lib/components/ui/EditText.svelte

```diff
@@ -0,0 +1,81 @@
+<script lang="ts">
+  import { Edit } from "@lucide/svelte";
+  import Input from "./Input.svelte";
+  import Button from "./Button.svelte";
+  import PlaceHolder from "./PlaceHolder.svelte";
+
+  type Props = {
+    text: string;
+    // onSave should handle the error.
+    onSave: (text: string) => Promise<void>;
+  };
+
+  let { text, onSave }: Props = $props();
+
+  let isEditing = $state(false);
+  let isLoading = $state(false);
+  let inputValue = $state(text);
+  let inputElement = $state<HTMLInputElement>();
+
+  const startEditing = () => {
+    inputValue = text;
+    isEditing = true;
+
+    // Focus the input after the DOM updates
+    setTimeout(() => {
+      if (inputElement) {
+        inputElement.focus();
+      }
+    }, 0);
+  };
+
+  const handleSave = async () => {
+    if (!isEditing || inputValue === text) {
+      isEditing = false;
+      return;
+    }
+
+    isEditing = false;
+    isLoading = true;
+    try {
+      await onSave(inputValue);
+    } finally {
+      isLoading = false;
+    }
+  };
+
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === "Enter") {
+      handleSave();
+    } else if (event.key === "Escape") {
+      inputValue = text;
+      isEditing = false;
+    }
+  };
+</script>
+
+<div class="inline-flex items-center gap-1">
+  {#if isEditing}
+    <Input
+      bind:element={inputElement}
+      bind:value={inputValue}
+      onblur={handleSave}
+      onkeydown={handleKeyDown}
+      type="text"
+      disabled={isLoading}
+      size="sm"
+    />
+  {:else if isLoading}
+    <PlaceHolder class="w-30" />
+  {:else}
+    <span class="name-display">{text}</span>
+    <Button
+      variant="tertiary"
+      iconOnly
+      onclick={startEditing}
+      aria-label="Edit name"
+    >
+      <Edit size={16} />
```

Comment:
Icons should scale with their button size based on screen scaling size instead of having an absolute size value.
```suggestion
      <Edit size={'1.25rem'} />
```